source/System.cpp

```diff
@@ -966,6 +974,13 @@ int System::MinimumFleetPeriod() const
 
 
 
+int System::LingerTime() const
+{
+	return lingerTime >= 0 ? lingerTime : minimumFleetPeriod / 4;
```

Comment:
we could also initialize it at the end of load so we dont have to calc this each time?

Reply:
No, if it is to be cached, it belongs in `UpdateSystem()`

```C++
// Update any information about the system that may have changed due to events,                                                                                                                                                                                                   
// or because the game was started, e.g. neighbors, solar wind and power, or                                                                                                                                                                                                      
// if the system is inhabited.                                                                                                                                                                                                                                                    
void System::UpdateSystem(const Set<System> &systems, const set<double> &neighborDistances)
```

Reply:
Okay, now I have:
- `requestedLingerTime` = the data file "linger time"
- `actualLingerTime` = a cached calculation in UpdateSystem()
The AI uses the actualLingerTime, of course.

Reply:
I mean since it was 4 you were dividing by it was prob fine-ish but good, every opti counts